plugins/inputs/slurm/README.md

```diff
@@ -0,0 +1,227 @@
+# SLURM Input Plugin
+
+This plugin gather diag, jobs, nodes, partitions and reservation metrics by
+leveraging SLURM's REST API as provided by the `slurmrestd` daemon.
+
+This plugin targets the `openapi/v0.0.38` OpenAPI plugin as defined in SLURM's
+documentation. That particular plugin should be configured when starting the
+`slurmrestd` daemon up. For more information, be sure to check SLURM's
+documentation [here][SLURM Doc].
+
+A great wealth of information can also be found on the repository of the
+Go module implementing the API client, [pcolladosoto/goslurm][].
+
+[SLURM Doc]: https://slurm.schedmd.com/rest.html
+[pcolladosoto/goslurm]: https://github.com/pcolladosoto/goslurm
+
+## Global configuration options <!-- @/docs/includes/plugin_config.md -->
+
+In addition to the plugin-specific configuration settings, plugins support
+additional global and plugin configuration settings. These settings are used to
+modify metrics, tags, and field or create aliases and configure ordering, etc.
+See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
+
+[CONFIGURATION.md]: ../../../docs/CONFIGURATION.md#plugins
+
+## Configuration
+
+```toml @sample.conf
+# Gather SLURM metrics
+[[inputs.slurm]]
+  ## Slurmrestd URL. Both http and https can be used as schemas.
+  url = "http://127.0.0.1:6820"
+
+  ## Credentials for JWT-based authentication.
+  # username = "foo"
+  # token = "topSecret"
+
+  ## Endpoints to skip gathering data from. By default all
+  ## the endpoints are queried for data.
+  # ignoreDiag = true
+  # ignoreJobs = true
+  # ignoreNodes = true
+  # ignorePartitions = true
+  # ignoreReservations = true
+
+  ## Maximum time to receive response.
+  # response_timeout = "5s"
+
+  ## Optional TLS Config. Note these options will only
+  ## be taken into account when the scheme specififed on
+  ## the URL parameter is https. They will be silently
+  ## ignored otherwise.
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+```
+
+## Metrics
+
+Given the great deal of metrics offered by SLURM's API, an attempt has been
+done to strike a balance between verbosity and usefulness in terms of the
+gathered information.
+
+- slurm_diag
+  - tags:
```

Comment:
This should have at least one tag to help determine what server it came from right in the event you have two of these plugins running? Like the url?

Should URL be a tag in all of these metrics?

Reply:
That's a great idea! Thanks a ton for the suggestion.

I have added the URL as a tag for every metric on 985a948660ca6081f0a8340fd0bb3e8520afaeae.

Thanks again!

Reply:
If I may add my remarks on this; The content in the tag `url` is not actually an URL, but just the hostname?

In that case we tend to use `source` as tag name, this in order to be more aligned with the other existing plugins.

Reply:
Yep, it's the hostname and not really a URL... Thanks a ton for the input!

I've renamed `url` to `source` on e4415ce19bdfa98614e1ae982d98cae0cdfbf9f3.

Thanks again!